crates/c_ext/src/sparse_observable.rs

```diff
@@ -10,130 +10,57 @@
 // copyright notice, and modified files need to carry a notice indicating
 // that they have been altered from the originals.
 
+use crate::exit_codes::ExitCode;
 use num_complex::Complex64;
-use qiskit_accelerate::sparse_observable::{BitTerm, SparseObservable, SparseTerm};
-
-/// @ingroup PauliTermVec
-/// A Pauli term vector, containing ``(index, bit_term)`` tuples.
-#[derive(Debug, Clone)]
-pub struct PauliTermVec {
-    // We store the data as two separate vectors, which simplified constructing
-    // SparseTerms later on, which expect this format.
-    indices: Vec<u32>,
-    bit_terms: Vec<BitTerm>,
+use qiskit_accelerate::sparse_observable::{BitTerm, SparseObservable, SparseTermView};
+use thiserror::Error;
+
+/// Errors related to C input.
+#[derive(Error, Debug)]
+pub enum CInputError {
+    #[error("Unexpected null pointer.")]
+    NullPointerError,
+    #[error("Non-aligned memory.")]
+    AlignmentError,
 }
```

Comment:
Would this make more sense to have in `exit_codes` or some error-specific module?
I suspect there will be more like this in the future and then it would be weird to import this from `sparse_observable` all the time.

Reply:
Good point, I can move this 👍🏻